Several updates (morbid map, labeling, genes)

joeflack4 opened the pull request:
### Updates
- Bugfix: morbidmap.txt parsing: Fixing various issues related to how we're parsing this and adding to the graph, now that we've discovered that several assumptions are false (e.g. oftentimes there is no phenotype MIM number).
- Updates: morbidmap.txt parser:
  - Now extracting phenotype mapping / association evidence information.
  - Also reified the way we're representing these relationships. No longer a single triple using a single RO relationship.
  - We're using other RO predicates now as well.
  - In the case of multiple mapping evidence, we're just using RO:0003302 (https://github.com/monarch-initiative/omim/issues/75).
  - In cases where no MIM# in Phenotype field, not adding an association.
  - In cases where the mapping key within the Phenotype field is '1', not adding an association (see: https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780)
- Bugfix: In some cases when there was not a label / alt label, 'None' was being added as the label.I
- Bugfix: Invalid xrefs were being added, e.g. OMIM:1 -> OMIM:14. This was an artefact of inconsistency in intra-label data model within morbidmap.txt's "Phenotype" field; that is, sometimes there was a missing MIM number, and in those cases, if there was an integer at the end of the phenotype label, it was being interpreted as a MIM number.
- Misc: (i) added some todo's, (ii) code comments, (iii) minor refactorings.
- Update: Added CONFIG var with 'verbose' as an option, defaulting to False. Now prints no INFO unless this is True.
- Add: additional report TSV specific to morbidmap.txt Phenotypes with no MIM#
- Add: rdfs:comment regarding the text ', INCLUDED' which appears in some labels.
- Rename: data/omim.ttl --> data/legacy_omim.ttl

### Addresses
- [x] #74
- [x] #67
- [x] #58
- [x] #8 
- [x] #75
- [x] #78
- [ ] #79
  - This one may remain open afterwards due to some follow-up considerations.

### Remaining sub-issues (#73)
- [x] 1. [How to add more info about mapping evidence](https://github.com/monarch-initiative/omim/pull/72#discussion_r1007503719)
- [x] 2. [Sometimes I shouldn't add association evidence triple?](https://github.com/monarch-initiative/omim/pull/72#discussion_r1007501794)

matentzn commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -254,14 +254,53 @@ def parse_morbid_map(lines) -> Dict[str, List[str]]:
-        gene_mim_number = tokens[2].strip()
-        cyto_location = tokens[3].strip()
+        gene_mim_number: int = tokens[2].strip()
+        gene_mim_number: str = str(gene_mim_number)
```
is this good programming practice (changing the datatype of a variable)? I am sure a linter will scream :P

joeflack4 replied:
The linter can kiss my butt, lol. I really do wish though that I felt less spread out and felt I had the time (or was disciplined enough) to have tests and lint / codestyle checkers passed on all my repos.

Yeah this is super ad hoc at this point. This is more a reminder to me that I would like to primarily treat this as `int` in the code, but right now the codebase is set up to read these MIM numbers as strings. I suppose adding another `todo` to the code would work; I tend to have them littered all over the place.

matentzn reviewed (commented):
Nothing yet to comment on it! I trust you with this more than myself.

joeflack4 commented:
@matentzn I'm gonna merge this and make a new release. I will probably re-run mondo-ingest pipeline tomorrow to use this. If you review and notice anything worth bringing up, let me know and I can make a subsequent PR w/ more changes if necessary.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -231,10 +225,14 @@ def run(use_cache: bool = False):
-        phenotype_mim_number, cyto_location = morbid_map[mim_number]
+    morbid_map: Dict[str, Dict] = parse_morbid_map(get_mim_file('morbidmap.txt', download_files_tf))
+    for mim_number, mim_data in morbid_map.items():
+        # todo?: unused `mim_data` keys:
```
```py
        # todo?: unused `mim_data` keys:
        #   'gene_mim_number': Should this be used somewhere?
        #   'gene_symbols':  Should this be used somewhere?
        #   'phenotype_label': I think this already added to graph because the MIM# and label shared in other file(s)
```

@matentzn `morbidmap.txt` has _phenotype/disorder MIM #'s_, and also _gene MIM#'s_. However, it looks like we have not been doing anything with the gene MIM number. I'm wondering if maybe Dazhi didn't do anything with this because, similar to the phenotype label, we're already extracting the MIM#, label, and information from one of the other files. I guess I could do with some double checking on this.

Now that I'm thinking about it, perhaps that's why Dazhi didn't do anything with the _gene symbols_ here as well. Perhaps that's because this is already being handled by the parsing of `mim2gene.txt` or elsewhere.

joeflack4 replied:
We spoke today and it looks like things are fine for now. This is probably correct, that this data is being fetched elsewhere.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -243,6 +241,10 @@ def run(use_cache: bool = False):
+        if mim_data['phenotype_mapping_info_label']:
+            # todo: store mim_data['phenotype_mapping_info_key'] in provenance somehow?
+            # https://biolink.github.io/biolink-model/docs/GeneToDiseaseAssociation.html
+            graph.add((OMIM[mim_number], BIOLINK['GeneToDiseaseAssociation'], BIOLINK['has_evidence']))
```
@cmungall @matentzn This is the new information that I'm adding which I was able to extract from `morbidmap.txt`, which probably fully addresses #73. I looked at the code that Kevin Schaper shared with me. His code used Koza to do add this information, but I believe the result is the same.

Example from new `omim.ttl`:
```ttl
OMIM:109480 a owl:Class ;
    rdfs:label "BSG" ;
   ...
    oboInOwl:hasExactSynonym "BSG",
        "basigin",
        "cd147 antigen",
        "extracellular matrix metalloproteinase inducer",
        "m6 leukocyte activation antigen",
        "tcsf" ;
   ...
    biolink:GeneToDiseaseAssociation biolink:has_evidence ;
    biolink:category biolink:Gene .
```

joeflack4 replied:
Discussed this with Nico, Kevin Schaper, and Sabrina today. Looks like `koza` is fetching mostly the same type of info here, except that there has been some configuration to load up some manually curated ECO mappings on each of the 4 OMIM phenotype mappings. Nico came to determination that I can add the literal text of the phenotype mapping descriptions, and that will be fine for now.

joeflack4 commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -18,9 +19,19 @@
-
-def retrieve_mim_file(file_name: str, download=False, return_df=False) -> Union[List[str], pd.DataFrame]:
+# MIM_NUMBER_DIGITS = 6  # if you see '6' in a regexp, this is what it refers to
+MORBIDMAP_PHENOTYPE_MAPPING_KEY_MEANINGS = {
```
@cmungall @matentzn If there is one of these mapping keys in a `morbidmap.txt` Phenotype label, we are now adding tripe (`MIM#` `biolink:GeneToDiseaseAssociation` `biolink:has_evidence`). However, I imagine we also want to add the kind of evidence itself? Not sure if we want to keep the "mapping key" (1 through 4), but the mapping key description, we should probably add some how. I imagine this involves me adding some kind of axiom with annotation properties? This a more advanced thing than I've typically done  in RDF. Wondering what you guys suggest?

joeflack4 replied:
Discussed this with Nico, Kevin Schaper, and Sabrina today. Looks like koza is fetching mostly the same type of info here, except that there has been some configuration to load up some manually curated ECO mappings on each of the 4 OMIM phenotype mappings. Nico came to determination that I can add the literal text of the phenotype mapping descriptions, and that will be fine for now.

joeflack4 commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -233,36 +245,48 @@ def parse_mim2gene(lines, filename='mim2gene.tsv', filename2='genemap2.tsv') ->
-    # }
-    ret = {}
-    p = re.compile(r".*,\s+(\d+)\s\(\d\)")
+def parse_morbid_map(lines) -> Dict[str, Dict]:
```
This is where most of the changes happened for #73. I got an updated regex [from Kevin's code](https://github.com/monarch-initiative/monarch-ingest/blob/main/monarch_ingest/ingests/omim/gene_to_disease.py) which provided a fix for #74.

matentzn commented on omim2obo/main.py:
```diff
@@ -180,7 +176,7 @@ def run(use_cache: bool = False):
             pass
 
-        if use_abbrev_over_label:
+        if use_abbrev_over_label and abbrev:
```
Makes sense!

matentzn commented on omim2obo/main.py:
```diff
@@ -243,6 +241,12 @@ def run(use_cache: bool = False):
+            #  we can do it. It would involve first the (OMIM[mim_number], BIOLINK['GeneToDiseaseAssociation'],
+            #  BIOLINK['has_evidence']), followed by some way to annotate w/ mim_data['phenotype_mapping_info_label']
+            #  https://biolink.github.io/biolink-model/docs/GeneToDiseaseAssociation.html
+            graph.add((OMIM[mim_number], BIOLINK['has_evidence'], Literal(mim_data['phenotype_mapping_info_label'])))
```
Related if you want to go down the SPARQL route: https://github.com/monarch-initiative/mondo-ingest/blob/0b255f5ded749a424fcc072444e672833875ed37/src/sparql/fix_illegal_punning_omim.ru

matentzn replied:
Also related to #60

joeflack4 replied:
These are great suggesitons. I would do this the Python rdflib route rather than SPARQL after the fact.

Before I make these changes, let's discuss further, in light of a few things:
1. How to add multiple gene::phenotype associations
2. What to do when there is no phenotype MIM#

joeflack4 replied:
Done! Example:
```owl
OMIM:100678 a owl:Class ;
    rdfs:label "ACAT2" ;
    ...
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ],
        SO:0000704 ;
    biolink:category biolink:Gene .
```

```owl
[] a owl:Axiom ;
    rdfs:comment "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:100678 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ] .
```

joeflack4 replied:
@matentzn Now, multiple gene::disease associations are being added in this way. I need you to check this:

Class snippet. Note that there are 3 identical subclass bnodes with refs to `OMIM:266300`:
```ttl
OMIM:155555 a owl:Class ;
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:203200 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:613099 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:613098 ],
        [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ],
```

For each of the 3 `OMIM:266300` refs above, there are 3 of these identical axioms now in the output:
```ttl
[] a owl:Axiom ;
    rdfs:comment "Evidence: (3) The molecular basis for the disorder is known; a mutation has been found in the gene." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:155555 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003302 ;
            owl:someValuesFrom OMIM:266300 ] ;
    biolink:has_evidence "Evidence: (3) The molecular basis for the disorder is known; a mutation has been found in the gene." .
```

## Questions
1. Do we want to add the `Phenotype` label so that these do not appear as duplicates? Note that they are not duplicates in `morbidmap.txt`. They just look like duplicates here because we leave out that `Phenotype` label, which differs between the 3. The labels are the only difference between these 3 instances.

Here are the original rows which include the labels:
```tsv
Phenotype	Gene Symbols	MIM Number	Cyto Location
[Skin/hair/eye pigmentation 2, blond hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
{UV-induced skin damage}, 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
```

If we leave out the label, I can dedupe these structures. It is lossy, but I don't know if that matters at this point.

matentzn replied:
- The fact that the exact same "axiom" can appear twice is just a consequence of rdflib - once this is cycled through ROBOT, which uses a set-based system for statements, these will all collapse into 1. No need to write code for deduplication!
- Can you remind me (sorry, too many things for my tiny brain) what the phrase `Do we want to add the Phenotype label?` means concretely? Can you give an example?

joeflack4 replied:
Ah, I see. So the OWL `.ttl` that's being output by `rdflib` is invalid. I could certainly tweak my code to avoid this duplication, but given that it's going through `robot`, let's handle it that way for now as you suggested, thanks!

---

Regarding "phenotype label", I can help elaborate on what I mean. Consider the second of the 3 rows above:
> `[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)`

This `Phenotype` field consists of 3 parts:
i. Label: `[Skin/hair/eye pigmentation 2, red hair/fair skin]`
ii. MIM#: `, 266300`
iii. Mapping evidence key: ` (3)`

At least, 'label' is what I call it. I suppose you could also call it a 'title' or 'name'.

joeflack4 replied:
Ah, I see. So the OWL `.ttl` that's being output by `rdflib` is invalid. I could certainly tweak my code to avoid this duplication, but given that it's going through `robot`, let's handle it that way for now as you suggested, thanks!

---

Regarding "phenotype label", I can help elaborate on what I mean. Consider the second of the 3 rows above:
> `[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)`

This `Phenotype` field consists of 3 parts:
i. Label: `[Skin/hair/eye pigmentation 2, red hair/fair skin]`
ii. MIM#: `, 266300`
iii. Mapping evidence key: ` (3)`

At least, 'label' is what I call it. I suppose you could also call it a 'title' or 'name'.

matentzn replied:
> Do we want to add the Phenotype label so that these do not appear as duplicates?

Ok now that I understand what you mean, i am confused: why would there ever be a different label for the same MIM number?

joeflack4 replied:
@matentzn You tell me Nico! @sabrinatoro @nicolevasilevsky Maybe you guys know: Why do we think that `morbidmap.txt` provides different labels for the same MIM# within the `Phenotype` column, as below?

```tsv
Phenotype	Gene Symbols	MIM Number	Cyto Location
[Skin/hair/eye pigmentation 2, blond hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
[Skin/hair/eye pigmentation 2, red hair/fair skin], 266300 (3)	MC1R, SHEP2, CMM5	155555	16q24.3
```

If we don't know the answer to this, I can send an email query to my OMIM contacts.

joeflack4 replied:
we have been discussing this in some issues. i dont remember at the moment what our planned action is though

joeflack4 replied:
the open issues for this have the apporpriate discussion on follow up steps

joeflack4 commented on omimIssues.json:
```diff
@@ -0,0 +1,40 @@
+{
```
If there are genuine issues with the latest release of `omim.ttl`, and I'm reporting them, I might as well commit the report as well. But I can always add this to the `.gitignore`.

joeflack4 replied:
I reported on these issues and no longer need this. Removed.

joeflack4 commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -233,36 +256,111 @@ def parse_mim2gene(lines, filename='mim2gene.tsv', filename2='genemap2.tsv') ->
+        # ...MIM number. In actuality, sometimes there is no MIM num, e.g.:
+        #   Phenotype	Gene Symbols	MIM Number	Cyto Location
+        #   {Autism susceptibility 14A} (2)	DEL16p11.2, C16DELp11.2, AUTS14A	611913	16p11.2
+        phenotype_identifier = phenotype_mim_number if phenotype_mim_number else phenotype_label
```
@matentzn I _really_ don't like this. I/we didn't realize this before, but what is inside the "Phenotype" column of `morbidmap.txt` is not a phenotype, but a "gene::phenotype association", with additional information (e.g. the mapping key, as we've discovered). This gene::phenotype association" has a label (what is shown in the "Phenotype" column is its label). There is sometimes (I previously thought "always") a phenotype MIM number inside of the "gene::phenotype association label". But the "gene::phenotype associations" themselves have no identifiers, other than the label (e.g. `{Autism susceptibility 14A} (2)` in this case).

```py
        # phenotype_identifier: By design, I would say that there should always be a numeric ID for this, such as a
        # ...MIM number. In actuality, sometimes there is no MIM num, e.g.:
        #   Phenotype	Gene Symbols	MIM Number	Cyto Location
        #   {Autism susceptibility 14A} (2)	DEL16p11.2, C16DELp11.2, AUTS14A	611913	16p11.2
        phenotype_identifier = phenotype_mim_number if phenotype_mim_number else phenotype_label
```

You can find a report of all of these such cases in the recent file I committed at the root of the repo: `omimIssues.json`

joeflack4 replied:
Marking this resolved as I made some issues where we'll carry on the discussion.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -231,18 +235,55 @@ def run(use_cache: bool = False):
+                # Report issue (No MIM # In Phenotype label), e.g.
+                #   Phenotype	Gene Symbols	MIM Number	Cyto Location
+                #   3p- syndrome (4)	DEL3pterp25, C3DELpterp25	613792	3pter-p25
+                if 'morbid_map' not in issues:
```
This is another case of ugly code where I'm checking for issues and reporting them. This will change once we've figured out what we want to do.

joeflack4 replied:
Done with reporting on these issues, removing this code.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -231,18 +235,55 @@ def run(use_cache: bool = False):
+        #  BIOLINK['has_evidence']), followed by some way to annotate w/ mim_data['phenotype_mapping_info_label']
+        #  https://biolink.github.io/biolink-model/docs/GeneToDiseaseAssociation.html
+        # graph.add((OMIM[mim_number], BIOLINK['has_evidence'], Literal(mim_data['phenotype_mapping_info_label'])))
+        graph.add((OMIM[mim_number], BIOLINK['GeneToDiseaseAssociation'],
```
I saw you're suggestion regarding changing `graph.add((OMIM[mim_number], BIOLINK['has_evidence'], Literal(mim_data['phenotype_mapping_info_label'])))` to something more appropriate. But before I proceed to do that, we should discuss all the issues and nuances with `morbidmap.txt`, as I suspect that your suggestion might change.

I'm temporarily dumping the associations as literal JSON text here so that we can look at the structure as an example.

joeflack4 replied:
We discussed and have a solution. Marking this resolved.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -231,18 +235,55 @@ def run(use_cache: bool = False):
+            pheno_id = assoc['phenotype_identifier']
+            try:
+                pheno_id = int(pheno_id)
+                graph.add((OMIM[mim_number], RO['0003303'], OMIM[pheno_id]))
```
As mentioned earlier, not every gene::phenotype association has a phenotype MIM number to add. I try to add it, but if it detects that there is no MIM number (e.g. it's not an `int`), it temporarily dumps information into a JSON string until we figure out what to do.

joeflack4 replied:
Opened an issue for this, so gonna mark this resolved and we can continue the discussion there: https://github.com/monarch-initiative/omim/issues/79

joeflack4 commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -233,36 +256,104 @@ def parse_mim2gene(lines, filename='mim2gene.tsv', filename2='genemap2.tsv') ->
+        })
+
+    # Resolve multiple entries
+    issues = {}
```
I made some issues for these. marking this resolved. can focus the discussion in these places:
- https://github.com/monarch-initiative/omim/issues/81
- https://github.com/monarch-initiative/omim/issues/78

joeflack4 commented:
@matentzn I think not. After our discussion on Friday and in this PR, I've added the following issues to be addressed by this PR (I've added a full list of related issues in the OP and in "Development" on the right-hand side of the PR page):
- #8 
- #78

I've put this back to a _draft_ for now. My goal is to complete these by Friday and have you review again.

joeflack4 commented:
@matentzn Ready again for review.
[omim.ttl.zip](https://github.com/monarch-initiative/omim/files/10036554/omim.ttl.zip)

matentzn commented on omim2obo/main.py:
```diff
@@ -231,18 +243,53 @@ def run(use_cache: bool = False):
+            b3 = BNode()
+            graph.add((b3, RDF['type'], OWL['Restriction']))
+            graph.add((b3, OWL['onProperty'], RO['0003303']))
+            if p_mim:
```
Dthis dpes not work correctly:

```
OMIM:101850 a owl:Class ;
    rdfs:label "palmoplantar keratoderma, punctate  iia 3" ;
    IAO:0000142 PMID:13116681,
        PMID:141935,
        PMID:4449867,
        PMID:4694520,
        PMID:6221990,
        PMID:8651714,
        PMID:9870680 ;
    RO:0002525 CHR:9606chr2p25-p12 ;
    oboInOwl:hasExactSynonym "PPKP3",
        "acrokeratoelastoidosis",
        "collagenous plaques of hands and feet",
        "palmoplantar keratoderma, punctate  iia 3" ;
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom "?Keratoderma, palmoplantar, punctate type 3" ] ;
    skos:exactMatch UMLS:C0545044 ;
    biolink:category biolink:Disease .
```

joeflack4 replied:
@matentzn I should have brought this up as a question to you today. 

`?Keratoderma, palmoplantar, punctate type 3` is one of those `Phenotype` in `morbidmap.txt` that does not have a MIM number. After discussion on Friday Nov 11, our decision was to add the relationship to the label, rather than the MIM number, that's what you see here. However, after our discussion today, Nov 18, we've learned that these rows with no MIM # are problematic parts of the data model. Exomiser skips. We will also skip for now until we know what to do with them.

So, I will update the code to not add these relationships / reifications in these cases.

joeflack4 replied:
`omim-temp.ttl`. Slipped in; will delete! Thanks for noticing.

joeflack4 replied:
`omim-temp.ttl` deleted, and the "something gone wrong" (as described above) are complete, so marking resolved.

matentzn commented on omim2obo/main.py:
```diff
@@ -231,18 +243,53 @@ def run(use_cache: bool = False):
+                graph.add((b3, OWL['someValuesFrom'], OMIM[p_mim]))
+            else:
+                graph.add((b3, OWL['someValuesFrom'], Literal(p_lab)))
+            graph.add((b2, OWL['annotatedTarget'], b3))
```
Stilling missing the evidence annotations on top?

joeflack4 replied:
@matentzn Actually, I'm not sure where you want me to add this. To the axiom, along with the comment about the evidence, or directly to the class. I will add it to the axiom since that is more consistent. So it will look like:

```ttl
OMIM:100678 a owl:Class ;
    rdfs:label "ACAT2" ;
    ...
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ],
        SO:0000704 ;
    biolink:category biolink:Gene .

[] a owl:Axiom ;
    biolink:has_evidence "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known."   # <------------------------ this is the new addition
    rdfs:comment "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:100678 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ] .
```

joeflack4 replied:
@matentzn Actually, I'm not sure where you want me to add this. To the axiom, along with the comment about the evidence, or directly to the class. I will add it to the axiom since that is more consistent. So it will look like:

```ttl
OMIM:100678 a owl:Class ;
    rdfs:label "ACAT2" ;
    ...
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ],
        SO:0000704 ;
    biolink:category biolink:Gene .

[] a owl:Axiom ;
    biolink:has_evidence "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known."   # <------------------------ this is the new addition
    rdfs:comment "Evidence: (1) The disorder is placed on the map based on its association with a gene, but the underlying defect is not known." ;
    owl:annotatedProperty rdfs:subClassOf ;
    owl:annotatedSource OMIM:100678 ;
    owl:annotatedTarget [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom OMIM:614055 ] .
```

matentzn replied:
Yes definitely to the axiom!

matentzn reviewed (changes requested):
- [x] Remove `omim-temp.ttl` from PR?
- [ ] Something gone wrong with OWL re-ification:

```ttl
OMIM:101850 a owl:Class ;
    rdfs:label "palmoplantar keratoderma, punctate  iia 3" ;
    rdfs:subClassOf [ a owl:Restriction ;
            owl:onProperty RO:0003303 ;
            owl:someValuesFrom "?Keratoderma, palmoplantar, punctate type 3" ] ;
    biolink:category biolink:Disease .
```

joeflack4 commented on omim2obo/parsers/omim_txt_parser.py:
```diff
@@ -12,22 +13,57 @@
+#  - https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780
+# MORBIDMAP_PHENOTYPE_MAPPING_KEY_PREDICATES provenance:
+# - https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780
+MORBIDMAP_PHENOTYPE_MAPPING_KEY_PREDICATES = {
```
@matentzn See my comment here, and Sabrina's above it: https://github.com/monarch-initiative/omim/issues/79#issuecomment-1320997956
For the purposes of merging this PR, I'm not sure if I should:
(a) continue just using `RO_0003303` for everything for now, as we were doing before
(b) Use different `RO` predicates based on Sabrina's default for each `morbidmap.txt` mapping key
(c) Use different `RO` predicates based _a number of different parameters_ as Sabrina discusses

Going with (b) for now until hearing otherwise.

joeflack4 commented on omim2obo/main.py:
```diff
@@ -230,19 +241,67 @@ def run(use_cache: bool = False):
+            # Provenance for this decision:
+            # - Multiple rows, same mapping key: https://github.com/monarch-initiative/omim/issues/75
+            # - Multiple rows, diff mapping keys: https://github.com/monarch-initiative/omim/issues/81
+            predicate = RO['0003302']  # default if `len(assocs) > 1`
```
@matentzn Just wanted to clarify I didn't misunderstand anything. See: https://github.com/monarch-initiative/omim/issues/75#issuecomment-1320995876

joeflack4 commented:
@matentzn This has been ready to merge for a while. I just went through again and verified evrything resolved.

joeflack4 commented:
@matentzn i just looked at my notes and i saw you'd already given me approval to merge this after doing an analysis on obsolete terms, which was successfully completed
